doc/source/user/basics.types.rst

```diff
@@ -12,87 +12,116 @@ Array types and conversions between types
 NumPy supports a much greater variety of numerical types than Python does.
 This section shows which are available, and how to modify an array's data-type.
 
-The primitive types supported are tied closely to those in C:
+NumPy's primitive types are tied closely to those in C:
```

Comment:
Then I think for `bool` would be that bool does not exists and `0` and `1` ints are preferred (or `<stdbool.h>` is required). 
For `intp`/`uintp` the same name could be considered as "C-like".
For `longlong`, because it only a C concept ported to Python?

Reply:
I think it's ok to use the `stdbool` types in the table, we're already using other c99 types.

I think I just wanted you to comment that some of the type names have a trailing underscore to avoid confusion with builtin python type names. No need to go into detail on longlong and intp.

Reply:
Right, I added a comment about trailing underscore.